### PR TITLE
Entry editing

### PR DESCRIPTION
**Pull request by modaralgayal**
The function missing is now fixed.

### CODE REVIEW

**Comment by JuhoPaananen** [comment]
Inputting wrong field name while editing crashes the app. There should be error handling for wrong inputs. 
image

**Comment by sevonj** [comment]
Could  possibly copy the delete selection dialog here.

File: src/ui.py [hunk 1]
```diff
@@ -329,3 +330,36 @@ def search_doi(io, app: App):
...
+        """Input the field that want to edit in the entry
+                        (author, title, journal, year, volume, number, pages): """
+    ).lower()
+
+    if field_to_edit == "author":
+        io.print(format_authors(entry_to_edit.persons.get("author")))
+    else:
+        io.print(entry_to_edit.fields[field_to_edit])
+
+    edited_field_value = io.input("Input the edition you want: ")
```
Comment (sevonj) [inline src/ui.py:359]: Some of these prints could be rephrased

Comment (sevonj) [inline src/ui.py:354]: Some of these prints could be rephrased.

Edit: Meant to select line 359 only.

File: src/ui.py [hunk 2]
```diff
@@ -329,3 +330,36 @@ def search_doi(io, app: App):
...
+
+    success = app.edit_entry(citekey, field_to_edit, edited_field_value)
+    if success:
+        io.print("Entry edited successfully!")
+    else:
+        io.print("Entry with the provided citekey was not found.")
```
Comment (sevonj) [inline src/ui.py]: This could be a loop, so that multiple fields could be changed in one "edit".

File: src/ui.py [hunk 3]
```diff
@@ -329,3 +330,36 @@ def search_doi(io, app: App):
...
+    if not citekey:
+        io.print("Citekey is missing, search cancelled.")
+        return
+    entry_to_edit = app.find_entries_by_citekey(citekey)
+    if not entry_to_edit:
+        io.print("No matching entry was found with the given Citekey.")
+        return
+    field_to_edit = io.input(
+        """Input the field that want to edit in the entry
+                        (author, title, journal, year, volume, number, pages): """
```
Comment (sevonj) [inline src/ui.py:351]: Could print the existing fields here instead of a hardcoded list.

Comment (sevonj) [reply src/ui.py:351]: Available fields tend to vary a lot.

Comment (sevonj) [reply src/ui.py:351]: Thinking further: Inserting a nonexistent field name should create said field.

**Pull request closed without merging**